### PR TITLE
Testing Youtube links for 2018 WTDAU

### PR DESCRIPTION
**Pull request by swapnilogale**

### CODE REVIEW

**Comment by plaindocs** [comment]
Looks good. I'll need to do some poking to actually add the Video pages though.

**Comment by swapnilogale** [comment]
@plaindocs - I'll work on this after we get the landing page for 2019 up and running

**Comment by plaindocs** [comment]
@swapnilogale you gotta update the master branch before before branching, or merge master in so you get the latest files.

File: README.md [hunk 1]
```diff
@@ -29,6 +29,50 @@ and have it be rendered properly at `2019`.
 ### Videos
 
 An even more fragile process which needs documenting and fixing.
+
```
Comment (plaindocs) [inline README.md:32]: @mxsasha @ericholscher FYI

**Comment by plaindocs** [comment]
@swapnilogale have a click around this and see if everything looks like it is in working order.

**Comment by swapnilogale** [comment]
@plaindocs - looks good. Thanks.

File: README.md [hunk 2]
```diff
@@ -29,6 +29,50 @@ and have it be rendered properly at `2019`.
...
+6. If you want to preview locally:
+
+    1. In `conf.py` around line 35 comment out the following two lines:
+
+       ```
+       if not on_rtd and not on_netlify and not on_travis:
+           exclude_patterns.append('videos')
+       ```
+
+    2. Run `make livehtml` and browse the new video pages at `http://127.0.0.1:8888`.
```
Comment (ericholscher) [inline README.md]:
```suggestion
    1. Run `BUILD_VIDEOS=True make livehtml` and browse the new video pages at `http://127.0.0.1:8888`.
```

File: README.md [hunk 3]
```diff
@@ -29,6 +29,50 @@ and have it be rendered properly at `2019`.
...
+1. In `_data/<year>.<city>.speakers.yaml`, add a `youtubeId: 12345678901` key value pair to each talk.
+
+2. Make sure the directory `videos/<city>/<year>` is included in the Video Archive `toctree` in `docs/videos/index.rst`.
+
+3. In `conf.py`, around line 130 uncomment the following two lines:
+
+   ```
+   #from _ext.videos import main
+   #html_context.update(main())
+   ```
```
Comment (ericholscher) [inline README.md]: We can definitely make this better --  we could even have it depend on the same logic around `BUILD_VIDEOS` to make this a simpler step.

**Review by ericholscher — APPROVED** [review]
Good to have this documented. Definitely something we should be able to improve with time.

File: README.md (hunk 3 above)
Comment (plaindocs) [reply README.md]: :100:  was thinking that after the recent convo around videos

**Comment by ericholscher** [comment]
Tried to cleanup the video generating tidbits here, hopefully it doesn't cause more issues :)

**Comment by plaindocs** [comment]
@ericholscher these improvements look great.

Looks like we're not checking for the has videos flag in the generation though, as there is a Portland 2019 error cropping up in the CI https://travis-ci.org/writethedocs/www/jobs/513732843#L1942

```
- /home/travis/build/writethedocs/www/.tox/links/tmp/html/videos/portland/2019/how-i-learned-to-stop-worrying-and-love-the-command-line-mike-jang/index.html
  *  internally linking to None, which does not exist (line 120)
     <a href="None" rel="external"><i class="fa fa-globe social-media-icon" aria-hidden="true"></i></a>
```

thoughts on an easy fix?

**Comment by plaindocs** [comment]
@ericholscher  I looked at adding 

```
            if year>2017 and 'flagvideos: False' in open('_data/config-'+region+'-'+str(year)+'.yaml').read():
                print('NO VIDEOS IN '+region+' '+str(year)+' yet.')
                continue
```

to `_ext/videos.py` around L62, but it isn't quite working. What am I missing?

**Review by plaindocs — APPROVED** [review]
Thanks! :-)

**Comment by ericholscher** [comment]
Hopefully it doesn't blow too much stuff up :)

**Comment by plaindocs** [comment]
:-D

**Comment by plaindocs** [comment]
Actually it does, it generates files for Vilnius which don't have videos.

**Comment by plaindocs** [comment]
But maybe, if we're thing of this as a talk archive, that is intended behavior?

**Comment by ericholscher** [comment]
Seems fine for now :)

**Comment by ericholscher** [comment]
We can update the URL's and whatnot over time (`/videos/` -> `/talks/`), but 👍 on talk archive that has video/slides/notes/photos/abstract across all conferences.

**Comment by plaindocs** [comment]
Sure. I'll open a PR with fixed up speaker.yaml and toc trees at some point. 

And yeah, :100: on talk archive over video archive.

**Pull request merged**